### PR TITLE
feat: AM-3270 MTLS for LDAP

### PR DESCRIPTION
**Pull request by lgw-gravitee**

### CODE REVIEW

**Comment by podlesrafal** [comment]
According to your description, there should be usage of `loadDefaultTrustStore` when variable is missing. I cannot see usage of this. Will it be used in the future or you missing this fallback in `load` method?

**Comment by lgw-gravitee** [comment]
the descrption is obsolete. It fallbacks to JVM truststore by default, so no need to code it on our side.

**Comment by podlesrafal** [comment]
So do we need `loadDefaultTrustStore()`? I cant find usage of it

**Comment by lgw-gravitee** [comment]
it is used in LDAP plugin

File: gravitee-am-certificate/gravitee-am-certificate-api/src/main/java/io/gravitee/am/certificate/api/DefaultTrustStoreProvider.java [hunk 1]
```diff
@@ -16,24 +16,19 @@
...
-
-    private DefaultTrustStoreProvider() {
-    }
+import lombok.experimental.UtilityClass;
 
-    public static String defaultTrustStorePassword() {
-        return System.getProperty(JAVA_DEFAULT_TRUSTSTORE_PASSWORD_PROPERTY);
-    }
+@UtilityClass
+public final class DefaultTrustStoreProvider {
```
Comment (podlesrafal) [inline gravitee-am-certificate/gravitee-am-certificate-api/src/main/java/io/gravitee/am/certificate/api/DefaultTrustStoreProvider.java:22]: using @UtilityClass, you dont need to set this class as final, as well as all fields as a static. https://projectlombok.org/features/experimental/UtilityClass

**Review by podlesrafal — APPROVED** [review]

File: gravitee-am-certificate/gravitee-am-certificate-api/src/main/java/io/gravitee/am/certificate/api/DefaultTrustStoreProvider.java (hunk 1 above)
Comment (lgw-gravitee) [reply gravitee-am-certificate/gravitee-am-certificate-api/src/main/java/io/gravitee/am/certificate/api/DefaultTrustStoreProvider.java:22]: agreed to leave as it is.

**Pull request merged**